### ISSUE TITLE
MobX conflicts

### ISSUE THREAD

**Issue by peterkle** [xogroup/union#290]
Context
===

- [ ] Visual bug?
- [x] Runtime bug?
- [x] Build bug?

### *package name(s) and versions*

`@xo-union/tk-component-membership-modal@0.1.0`

### *javascript runtime*

Our app uses MobX version 3.1.16
`tk-component-smart-form@0.1.0` uses version 3.2.2

Issue
===

### What are you trying to achieve or the steps to reproduce ?

When typing text into the Login/Signup modal, the text field does not update.

Remove the MobX package from `tk-component-smart-form@0.1.0` node modules, and it works.

### What result did you observe ?

![screen shot 2017-08-15 at 1 14 09 pm](https://user-images.githubusercontent.com/486241/29327014-b385d250-81bb-11e7-8aaa-9f64e79f939e.png)

**Comment by GeorgeTaveras1231** [comment]
@peterkle As I mentioned, this is an issue when the versions are not compatible.

FYI for anyone interested, what caused this issue is the fact that @peterkle's `package.json` file had a mobx dependency defined as: `"mobx": "3.1.6"` (no caret 🥕 ) which is not compatible with ours: `"mobx": "^3.2.2". 

A temporary fix for this problem could be to line up your `"mobx"` version with ours by changing your `package.json` file.

A better long term solution would be to define that as a `peerDependency`. 

@marioharper Going back to our conversation about the usage of `peerDependency`. It may make sense to establish a guideline for "When to use `peerDependency`s". Even though as I mentioned to you, I believe we should use `dependecy` whenever we explicitly depend on the API of the package, scenarios like this where two versions of the same package cause some significant problems in the parent application may be a use case to use `peerDependency`s

**Comment by marioharper** [comment]
Yeah this is also my concern with us having `react` as a `dependency` since two version of react being loaded can cause some super weird errors that don't always tell you explicitly that two versions being loaded are the cause. Same thing happened here. There was nothing obvious telling @peterkle that the two versions were the issue. Using `peerDependency` gives a better dev experience in this case.
